src/OpenTelemetry/BatchExportProcessorOptions.cs

```diff
@@ -14,11 +14,57 @@
 // limitations under the License.
 // </copyright>
 
+using System;
+using System.Security;
+using OpenTelemetry.Internal;
+
 namespace OpenTelemetry
 {
     public class BatchExportProcessorOptions<T>
         where T : class
     {
+        internal const string MaxQueueSizeEnvVarKey = "OTEL_BSP_MAX_QUEUE_SIZE";
+
+        internal const string MaxExportBatchSizeEnvVarKey = "OTEL_BSP_MAX_EXPORT_BATCH_SIZE";
+
+        internal const string ExporterTimeoutEnvVarKey = "OTEL_BSP_EXPORT_TIMEOUT";
+
+        internal const string ScheduledDelayEnvVarKey = "OTEL_BSP_SCHEDULE_DELAY";
+
+        public BatchExportProcessorOptions()
+        {
+            try
+            {
+                int value;
+
+                if (TryLoadEnvVarInt(BatchExportProcessorOptions<T>.ExporterTimeoutEnvVarKey, out value))
+                {
+                    this.ExporterTimeoutMilliseconds = value;
+                }
+
+                if (TryLoadEnvVarInt(BatchExportProcessorOptions<T>.MaxExportBatchSizeEnvVarKey, out value))
+                {
+                    this.MaxExportBatchSize = value;
+                }
+
+                if (TryLoadEnvVarInt(BatchExportProcessorOptions<T>.MaxQueueSizeEnvVarKey, out value))
+                {
+                    this.MaxQueueSize = value;
+                }
+
+                if (TryLoadEnvVarInt(BatchExportProcessorOptions<T>.ScheduledDelayEnvVarKey, out value))
+                {
+                    this.ScheduledDelayMilliseconds = value;
+                }
+            }
+            catch (SecurityException ex)
+            {
+                // The caller does not have the required permission to
+                // retrieve the value of an environment variable from the current process.
+                OpenTelemetrySdkEventSource.Log.MissingPermissionsToReadEnvironmentVariable(ex);
```

Comment:
I think this brings the question - do we expect it to be transactional? Either read all environment variables or none.

Reply:
Rather not. I will change it and also fix the same issue in other places.

Reply:
Addressed for `BatchExportProcessorOptions` here https://github.com/open-telemetry/opentelemetry-dotnet/pull/2219/commits/f679a640dcc9e6a197fb94a7aba46575a2aa63f7

I would rather address it for `OtlpExporterOptions` and `JaegerExporterOptions` together with https://github.com/open-telemetry/opentelemetry-dotnet/pull/2219#discussion_r681164662 in a dedicated PR.